docsrc/imap/reference/manpages/systemcommands/reconstruct.rst

```diff
@@ -162,6 +164,12 @@ Options
 
     Instead of mailbox prefixes, give usernames on the command line
 
+.. option:: -P
+
+    Instead of mailbox prefixes, give paths to cyrus.header files on
+    the command line.  This will ONLY create/repair mailboxes.db
+    records using data in cyrus.header and cyrus.index.
+
```

Comment:
it is unclear, whether paths will be scanned recursively, or `-P` has to be supplied many times, with a different path each time, or many calls to `reconstruct -P` shall be done for each path. 

That is, if I want to repair all mailboxes inder `/X`, can I call `reconstruct -P /X`, or do I have to use instead `for i in $(find /X); do reconstruct -P $i; done`.

Ideally there shall be an example for this.

Reply:
You can specify multiple paths to a single `reconstruct -P` and each is dealt with, but it doesn't recurse.

So `reconstruct -P /path/to/mailbox1/cyrus.header /path/to/mailbox2/cyrus.header` would work, as would something like `find /path -name cyrus.header | xargs reconstruct -P`.  But `reconstruct -P /path/to/mailbox1/cyrus.header` will not look for or find `/path/to/mailbox1/child/cyrus.header`.

Since the -P arguments are paths to cyrus.header files, I would be surprised by it trying to recurse.  The use-case here is that things are already damaged enough that it can't just find the cyrus.header by the usual means, so it should do exactly what it's told and no more.

If it did recurse, the recursion would only be able to work for legacy mailboxes, since uuid mailboxes aren't stored according to their mailbox hierarchy.

Reply:
> Since the -P arguments are paths to cyrus.header files

This is not spelled clear enough.  My understanding was that the parameter is a directory, containing a cyrus.header file.

Reply:
It can be a directory containing a cyrus.header or an explicit cyrus.header filename

Reply:
> It can be a directory containing a cyrus.header or an explicit cyrus.header filename

This needs to be spelled explicitly, otherwise the documentation is ambiguous, leading to mistrust.

Reply:
Fixed